workflow/executor.py

```diff
@@ -590,18 +593,23 @@ def run(self):
             try:
                 if step.engine_size:
                     self.resource_manager.add_engine(step.engine_size)
-                    step.execute(self.logger, self.config.env, self.resource_manager.get_rai_config(step.engine_size))
+                    self.execute_step(step, self.resource_manager.get_rai_config(step.engine_size))
                     next_step = steps_iter.peek(None)
                     if next_step and next_step.engine_size != step.engine_size:
                         self.resource_manager.remove_engine(step.engine_size)
                 else:
-                    step.execute(self.logger, self.config.env, rai_config)
+                    self.execute_step(step, rai_config)
 
                 end_time = time.time()
                 execution_time = end_time - start_time
                 query = "\n".join([q.update_step_state(step.idt, WorkflowStepState.SUCCESS.name),
                                    q.update_execution_time(step.idt, execution_time)])
                 rai.execute_query(self.logger, rai_config, self.config.env, query, readonly=False)
+            except StepTimeOutException as e:
+                # Skip step state update since write txn can be stuck and RWM will raise ConcurrentWriteAttemptException
+                if step.engine_size:
+                    self.resource_manager.remove_engine(step.engine_size)
```

Comment:
Let's explicitly document this param would DEPROVISION the engine in case a step timed out

Reply:
this is only for engines defined in steps

Reply:
so otherwise we'd just fail and the engine would be left around spinning? what if `--cleanup-enigne` passed?

Reply:
`--cleanup-engine` is covered on CLI level